vllm/v1/attention/backends/mamba2_attn.py

```diff
@@ -68,6 +74,40 @@ def __init__(self, kv_cache_spec: AttentionSpec, layer_names: list[str],
         self.chunk_size = vllm_config.model_config.get_mamba_chunk_size()
         assert self.chunk_size is not None, (
             "chunk_size needs to be set in the model config for Mamba2 models")
+        assert isinstance(kv_cache_spec, MambaSpec)
+        if kv_cache_spec.enable_prefix_caching:
+            self.state_indices_tensor = torch.empty(
+                (self.decode_cudagraph_max_bs,
+                 cdiv(vllm_config.model_config.max_model_len,
+                      kv_cache_spec.block_size)),
+                dtype=torch.int32,
+                device=device,
+            )
+            self.current_last_token_block_idx = torch.empty(
+                (self.decode_cudagraph_max_bs, ),
+                dtype=torch.int32,
+                device=device,
+            )
```

Comment:
If some of these tensors are only used for prefill, we actually don't need to do this stuff. We could simply create them each time in the builder.

Reply:
Done